src/controllers/uploadController.js

```diff
@@ -106,25 +121,82 @@ class UploadController extends PageController {
     return validationResult ? !validationResult.error : false
   }
 
-  // this function is a validation function that is called by the form wizard
-  static validateFileType ({ originalname }) {
-    const allowedFiletypes = [
-      'csv',
-      'xls',
-      'xlsx',
-      'json',
-      'geojson',
-      'gml',
-      'gpkg'
-    ]
-    // check file type
-    const fileType = originalname.split('.').pop()
-    if (!allowedFiletypes.includes(fileType)) {
+  static extensionIsValid (datafile) {
+    const allowedExtensions = ['csv', 'xls', 'xlsx', 'json', 'geojson', 'gml', 'gpkg']
+
+    const parts = datafile.originalname.split('.')
+
+    const extension = parts[parts.length - 1]
+    if (!allowedExtensions.includes(extension)) {
+      return false
+    }
+
+    return true
+  }
+
+  static sizeIsValid (datafile) {
+    const maxSize = 10 * 1024 * 1024 // 10MB
+
+    if (datafile.size > maxSize) {
+      return false
+    }
+
+    return true
+  }
+
+  static fileNameIsntTooLong (datafile) {
+    const maxSize = 255 // Maximum filename size
+    if (datafile.originalname.length > maxSize) {
+      return false
+    }
+    return true
+  }
+
+  static fileNameIsValid (datafile) {
+    const invalidCharacters = /[<>:"/\\|?*]/
+    if (invalidCharacters.test(datafile.originalname)) {
       return false
     }
     return true
   }
 
+  static fileNameDoesntContainDoubleExtension (datafile) {
+    const parts = datafile.originalname.split('.')
+    if (parts.length > 2) {
```

Comment:
Could this check be done at the same time as the "extensionIsValid" one? just copy this if  to the top of that function.